llama_index/llms/konko.py

```diff
@@ -130,12 +126,50 @@ def _get_model_name(self) -> str:
     def class_name(cls) -> str:
         return "Konko_LLM"
 
+    def _create_model_context_length_dict(self) -> dict:
+        try:
+            import konko
+        except ImportError:
+            raise ValueError(
+                "Could not import konko python package. "
+                "Please install it with `pip install konko`."
+            )
+
+        model_context_dict = {}
+        if is_openai_v1():
+            models = konko.models.list().data
+            for model in models:
+                model_name = model.name
+                max_context_length = model.max_context_length
+                model_context_dict[model_name] = max_context_length
+        else:
+            models = konko.Model.list().data
+            for model in models:
+                model_name = model["name"]
+                max_context_length = model["max_context_length"]
+                model_context_dict[model_name] = max_context_length
+
+        return model_context_dict
+
+    # Calculate the maximum number of tokens possible for a model
+    def _konko_modelname_to_contextsize(self, modelname: str) -> int:
+        all_available_models = self._create_model_context_length_dict()
+        context_size = all_available_models.get(modelname, None)
+
+        if context_size is None:
+            raise ValueError(
+                f"Unknown model: {modelname}. Please provide a valid Konko model name."
+                "Known models are: " + ", ".join(all_available_models.keys())
+            )
+
+        return context_size
+
     @property
     def metadata(self) -> LLMMetadata:
         return LLMMetadata(
-            context_window=konko_modelname_to_contextsize(self._get_model_name()),
+            context_window=self._konko_modelname_to_contextsize(self._get_model_name()),
             num_output=self.max_tokens,
-            is_chat_model=True,
+            is_chat_model=is_chat_model(self._get_model_name()),
```

Comment:
I think we may be doing some double work here: he request /v1/models once in `_konko_modelname_to_contextsize`, and then again in `is_chat_model`. we could easily turn this into a single request. we could either expand the dict we return in `_create_model_context_length_dict` or even better introduce a dataclass that would hold all relevant attributes of a model, and many such instances would be stored in the resulting dict.

Reply:
Makes sense! Did the changes

Reply:
hm, not sure I see them.

Reply:
Just pushed the code.